galaxy_skeleton2.R

```diff
@@ -1,106 +1,39 @@
-# tidyverse), but there is a delicate detail here. The "|"-character has special
-# meaning in R ("or"), so it must be *escaped*, meaning that we tell R that it
-# should be interpreted as a normal character. We do that by adding two forward
-# slashes in front of it. This function returns a list, with one element for
-# each input element. We only send one string in, and hence get only one list
-# element out (check that!). We just unlist it to get out the vector. Then we
-# apply the str_trim()-function (also in the stringr-package) to get rid of all
-# the empty space. Replace the question mark below:
-variable_names <- 
-  str_split(string = ?, pattern = "\\|") %>% 
-  unlist() %>% 
-  str_trim()
-
-# Read the data. One way to do this is to rewrite the data to a new .csv-file
-# with comma-separators for instance using cat() again, with the variable names
-# from the step above on the first line (see for instance paste() for collapsing
-# that vector to a single string with commas as separators).
-
-# Let us try the approach described above. It is quite transparent, but could
-# probably be done quicker. We take the elements in "raw_file" containing data,
-# replace all "|" with "," and remove all empty space. The gsub-function is
-# super for this kind of search-and-replace. Replace the question mark below.
-
-comma_separated_values <- 
-  ? %>% 
-  gsub("\\|", ",", .) %>% 
-  gsub(" ", "", .)
+library(tidyverse)
+
+raw_file <- readLines(con = "suites_dw_Table1.txt")
+
+L <- which(substr(x = raw_file, start = 1, stop = 2) == "--")[1]
 
-# We then just add the variable names (separated with commas) on top, and
-# cat()-the whole ting to a .csv-file in the same way as we did with the
-# variable descriptions above.
+cat(raw_file[1:(L-2)], sep = "\n", file = "variable_descriptions.txt")
 
-comma_separated_values_with_names <- 
-  c(paste(variable_names, collapse = ","),
-    comma_separated_values)    
+variable_names <- str_split(string = raw_file[L-1], pattern = "\\|") %>% unlist() %>% str_trim()
+
+comma_separated_values <- raw_file[(L+1):length(raw_file)] %>%
+  gsub("\\|", ",", .) %>%
+  gsub(" ", "", .)
 
-# Replace the question mark and come up with a file name
-cat(?, sep = "\n", file = "?")
+comma_separated_values_with_names <- c(paste(variable_names, collapse = ","), comma_separated_values)
 
-# Read the file back in as a normal csv-file. The readr-package is part of
-# tidyverse, so it is already loaded.
-galaxies <- read_csv("?")
+new_csv_file <- "galaxies_cleaned.csv"
 
+cat(comma_separated_values_with_names, sep = "\n", file = new_csv_file)
 
-# You should now have a nice, clean data frame with galaxies and their
-# characteristics in memory. As of March 2022 it should contain 796
-# observations.
+galaxies <- read_csv(new_csv_file)
 
+# Problem 3: Plot
 
+# I do not fully understand what the values in the dataframe means.Therefore I did not manage to complete this problem.
```

Comment:
Here, you could have plotted a scatterplot of distance vs mass (variable log_lk). In the plot, you would see a tendency that small galaxies are spotted more often in a distance close to us. This may indicate that it is harder to spot small galaxies far away from us.